plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -603,6 +604,17 @@ public function search_orders( $term ) {
 							WHERE order_item_name LIKE %s",
 							'%' . $wpdb->esc_like( wc_clean( $term ) ) . '%'
 						)
+					),
+					$wpdb->get_col(
+						$wpdb->prepare(
+							"SELECT DISTINCT s.order_id FROM {$wpdb->prefix}wc_order_stats s WHERE s.customer_id = %s",
+							$wpdb->get_col(
```

Comment:
`wc_order_stats.customer_id` and `wp_usermeta.user_id` are totally different things so we should not use them as a reference of each other. When there is no registered user (`Guest`) assigned to the order then `wc_order_stats.customer_id` will have a non NULL value and `wc_order_stats.customer_id` == `wp_usermeta.customer_id` but `wp_usermeta.user_id` will have NULL as value because `Guest` is a valid customer but not a registered user.

The relation between the `wc_order_stats` and the `wp_usermeta` tables could be done through the table `wp_wc_customer_lookup` so this query will looks like:

```sql
SELECT DISTINCT os.`order_id` FROM `wp_wc_order_stats` os
INNER JOIN `wp_wc_customer_lookup` cl ON os.`customer_id` = cl.`customer_id`
INNER JOIN `wp_usermeta` um ON cl.`user_id` = um.`user_id`
WHERE (um.`meta_key` = 'billing_phone' OR um.`meta_key` = 'shipping_phone') 
AND um.`meta_value` = {phoneNumber};
```

Reply:
@mdperez86 

Thank you for the details. 

I just updated the query, as you mentioned.